### PR TITLE
Add end to end tests that invoke psalm as a separate process

### PR DESCRIPTION
**Pull request by bdsl**

### CODE REVIEW

**Comment by bdsl** [comment]
This adds almost one second to the test time: https://298-184879327-gh.circle-artifacts.com/0/home/docker/project/build/phpunit/phpunit.xml

**Comment by bdsl** [comment]
Idea for this was mentioned in #1788 .

File: tests/EndToEnd/PsalmEndToEndTest.php [hunk 1]
```diff
@@ -0,0 +1,150 @@
+<?php
+
+
+namespace Psalm\Tests\EndToEnd;
+
+use PHPUnit\Framework\TestCase;
+
```
Comment (bdsl) [inline tests/EndToEnd/PsalmEndToEndTest.php]:
```suggestion
```

**Comment by muglug** [comment]
Thanks!

**Pull request merged**